internal/errorlint/errorlint.go

```diff
@@ -0,0 +1,191 @@
+// Copyright 2023 Outreach Corporation. All Rights Reserved.
+
+// Description: See package comment for this one file package.
+
+// Package errorlint contains the necessary logic for the error/log/trace linter. The error
+// errors.New, errors.WithMessage, errors.WithMessagef, trace.StartCall, trace.StartSpan, log.Warn,
+// log.Error, log.Info, errors.Wrap, errors.Wrapf, fmt.Errorf, errors.Errorf static messages for
+// capitalization. It also checks messages ending in punctuation ". : ! and \n"
+package errorlint
+
+import (
+	"go/ast"
+	"go/token"
+	"strconv"
+	"unicode"
+	"unicode/utf8"
+
+	"github.com/getoutreach/lintroller/internal/common"
+	"github.com/getoutreach/lintroller/internal/reporter"
+	"golang.org/x/tools/go/analysis"
+)
+
+// name defines the name for the errorlint linter.
+const name = "errorlint"
+
+// doc defines the help text for the error linter.
+const doc = `Ensures that each static message in error/trace/log/ is lowercase.
+
+A valid example is the following:
+
+	errors.New("org not found in launchdarkly rule")
+An invalid example is the following:
+
+	errors.New("Org not found in LAUNCHDARKLY rule")
+An invalid example is the following:
+
+	errors.New("Org not found in launchdarkly rule:")`
+
+// Analyzer exports the errorlint analyzer (linter).
+var Analyzer = analysis.Analyzer{
+	Name: name,
+	Doc:  doc,
+	Run:  errorlint,
+}
+
+// file represents a file being linted.
+type file struct {
+	f *ast.File
+}
+
+func (file *file) walk(fn func(ast.Node) bool) {
+	ast.Walk(walker(fn), file.f)
+}
+
+// walker adapts a function to satisfy the ast.Visitor interface.
+// The function returns whether the walk should proceed into the node's children.
+type walker func(ast.Node) bool
+
+func (w walker) Visit(node ast.Node) ast.Visitor {
+	if w(node) {
+		return w
+	}
+
+	return nil
+}
```

Comment:
You probably don't need to do this yourself and maintain the code - look at [ast.Inspect](https://pkg.go.dev/go/ast#Inspect). Here is an example usage within the lintroller codebase too - https://github.com/getoutreach/lintroller/blob/main/internal/doculint/doculint.go#L176

Reply:
added changes